code/game/machinery/computer/aifixer.dm

```diff
@@ -70,34 +71,40 @@
 	popup.open()
 	return
 
+/obj/machinery/computer/aifixer/proc/Fix()
+	. = TRUE
+	if(world.time < fix_next_tick)
+		return
+	while(. && world.time < fix_next_tick)

```

Comment:
Why not scale healing/powerdrain it to process wait argument instead of using weird while constructs.